routing/DELEGATED_CONTENT_ROUTING_HTTP.md

```diff
@@ -0,0 +1,99 @@
+
+# API Specification
+The Delegated Content Routing Routing HTTP API uses the `application/json` content type by default. Clients and servers *should* support `application/cbor`, which can be negotiated using the standard `Accept` and `Content-Type` headers.
+
+## Common Data Types:
+
+- CIDs are always encoded using a [multibase](https://github.com/multiformats/multibase)-encoded [CIDv1](https://github.com/multiformats/cid#cidv1).
+- Multiaddrs are encoded according to the [human-readable multiaddr specification](https://github.com/multiformats/multiaddr#specification)
+- Peer IDs are encoded according [PeerID string representation specification](https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md#string-representation)
+- Multibase bytes are encoded according to [the Multibase spec](https://github.com/multiformats/multibase), and *should* use Base64.
+- Timestamps are Unix millisecond epoch timestamps
+
+## API
+- `GET /v1/providers/{CID}`
+    - Response
+        
+        ```json
+        {
+            "Providers": [
+                {
+					"Peer": {
+						"ID": "12D3K...",
+						"Multiaddrs": ["/ip4/.../tcp/.../p2p/...", "/ip4/..."]
+					}
+                    "Protocols": [
+                        {
+                            "Codec": 2320,
+                            "Payload": { ... }
+                        }
+                    ]
+                }
+            ]
+        }
+        ```
+        
+    - Default limit: 100 providers
+    - Optional query parameters
+        - `transfer` only return providers who support the passed transfer protocols, expressed as a comma-separated list of [multicodec codes](https://github.com/multiformats/multicodec/blob/master/table.csv) in decimal form such as `2304,2320`
+        - `transport` only return providers whose published multiaddrs explicitly support the passed transport protocols, such as `460,478` (`/quic` and `/tls/ws`)
+        - Servers should treat the multicodec codes used in the `transfer` and `transport` parameters as opaque, and not validate them, for forwards compatibility
+- `GET /v1/providers/hashed/{multihash}`
+    - This is the same as `GET /v1/providers/{CID}`, but takes a hashed CID encoded as a [multihash](https://github.com/multiformats/multihash/)
+- `PUT /v1/providers`
+    - Reframe equivalent: Provide
+    - Body
+        ```json
+        {
+            "Signature": "multibase bytes",
+            "Payload": "multibase bytes"
+        }
+        ```
+	    - `Payload` is a multibase-encoded string containing a JSON object with the following schema:
+		```json
+		{
+                "Keys": ["cid1", "cid2"],
+                "Timestamp": 1234,
+                "AdvisoryTTL": 1234,
+                "Provider": {
+					"Peer": {
+						"ID": "12D3K...",
+						"Multiaddrs": ["/ip4/.../tcp/.../p2p/...", "/ip4/..."]
+					}
+                    "Protocols": [
+                        {
+                            "Codec": 1234,
```

Comment:
your most recent [comment](https://github.com/ipfs/specs/pull/337/files#r1017006162) has an example with codec being specified as a string quoted number. here it's directly a number. which one do we want to use?

Reply:
Yes due to that schema change and switching to tagged unions, I changed it to be a string quoted number so that it would be compatible with OpenAPI discriminators which requires that the value of the property referred to by `propertyName` be a string (see https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/). I'll post a revision soon that cleans all of this up (and addresses all the other comments).